src/Pact/Typechecker.hs

```diff
@@ -897,21 +910,9 @@ toAST TModRef{..} = do
 
   n <- trackNode ty tcid
   return $ ModRef n (_modRefName _tModRef) (_modRefSpec _tModRef)
-toAST (TVar v i) = case v of -- value position only, TApp has its own resolver
+toAST (TVar v _i) = case v of -- value position only, TApp has its own resolver
   (Left (Ref r)) -> toAST (fmap Left r)
-  (Left (Direct t)) ->
-    case t of
-      TLiteral {..} ->
-        -- Handle references to pre-evaluated constants:
-        trackPrim _tInfo (litToPrim _tLiteral) (PrimLit _tLiteral)
-      TConst{..} -> case _tModule of
-        -- if modulename is nothing, it's a builtin
-        Nothing -> toAST $ return $ Left (Direct $ constTerm _tConstVal)
-        _ -> die i $ "Non-native constant value in native context: " <> showPretty t
```

Comment:
Note, this enforcement is gone now. There's no real need for it -- even if it should be syntactically impossible for a user `defconst` to be `Direct`, it doesn't really mean anything is un-TC-able